src/oneD/StFlow.cpp

```diff
@@ -20,15 +20,15 @@ StFlow::StFlow(ThermoPhase* ph, size_t nsp, size_t points) :
     Domain1D(nsp+c_offset_Y, points),
     m_nsp(nsp)
 {
-    if (ph->type() == "ideal-gas") {
-        m_thermo = static_cast<IdealGasPhase*>(ph);
-    } else {
-        throw CanteraError("StFlow::StFlow",
-                           "Unsupported phase type: need 'IdealGasPhase'");
-    }
     m_type = cFlowType;
     m_points = points;
 
+    if (ph->type() == "ideal-gas" || ph->type() == "Redlich-Kwong" || ph->type() == "Peng-Robinson") {
+        m_thermo = static_cast<ThermoPhase*>(ph);
```

Comment:
The signature of the function indicates that this already is the correct type, so I'm not sure that the cast is necessary?

Reply:
Yup. I have removed it now.